cmd/nvidia-ctk-installer/container/runtime/containerd/config_test.go

```diff
@@ -0,0 +1,577 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+*/
+
+package containerd
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	cli "github.com/urfave/cli/v3"
+
+	"github.com/NVIDIA/nvidia-container-toolkit/cmd/nvidia-ctk-installer/container"
+	"github.com/NVIDIA/nvidia-container-toolkit/pkg/config/engine/containerd"
+	"github.com/NVIDIA/nvidia-container-toolkit/pkg/config/toml"
+)
+
+// TestContainerdConfigLifecycle tests the complete Setup->Cleanup lifecycle for both v1 and v2 configs.
+func TestContainerdConfigLifecycle(t *testing.T) {
+	c := &cli.Command{
+		Name: "test",
+	}
+
+	testCases := []struct {
+		description                 string
+		configVersion               int  // 0 for auto-detect, 1 for v1, 2 for v2
+		useLegacyConfig             bool // for v1 legacy behavior
+		containerOptions            container.Options
+		options                     Options
+		prepareEnvironment          func(*testing.T, string) error
+		expectedSetupError          error
+		assertSetupPostConditions   func(*testing.T, string) error
+		expectedCleanupError        error
+		assertCleanupPostConditions func(*testing.T, string) error
+	}{
+		{
+			description:   "v2: top-level config does not exist",
+			configVersion: 2,
+			containerOptions: container.Options{
+				Config:       "{{ .testRoot }}/etc/containerd/config.toml",
+				RuntimeName:  "nvidia",
+				RuntimeDir:   "/usr/bin",
+				SetAsDefault: false,
+				RestartMode:  "none",
+			},
+			options: Options{},
+			assertSetupPostConditions: func(t *testing.T, testRoot string) error {
+				require.FileExists(t, filepath.Join(testRoot, "etc/containerd/config.toml"))
+				verifyRuntimesPresent(t, filepath.Join(testRoot, "etc/containerd/config.toml"),
+					"nvidia", "nvidia-cdi", "nvidia-legacy")
+				return nil
+			},
+			assertCleanupPostConditions: func(t *testing.T, testRoot string) error {
+				configPath := filepath.Join(testRoot, "etc/containerd/config.toml")
+
+				if _, err := os.Stat(configPath); err == nil {
+					// TODO(@ArangoGutierrez): When moving to DropIn files, we should verify that the config file is deleted after cleanup.
+					// For now, we just verify that the runtimes are absent.
+					verifyRuntimesAbsent(t, configPath, "nvidia", "nvidia-cdi", "nvidia-legacy")
+				}
+				return nil
+			},
+		},
+		{
+			description:   "v2: existing config without nvidia runtime and CDI enabled",
```

Comment:
The case where `runc` or a default runtime is defined in the config is quite a different one to no default being defined. We may want to distinguish between these.